### PR TITLE
Add e2e test for oc downloads links

### PR DESCRIPTION
**Pull request by benjaminapetersen**
```bash
go test -run TestDownloadsEndpoint -count 1
checking for console-operator availability
found operator deployment
link for Download oc for Linux 200 OK (https://console-openshift-console.apps.bpetersen.devcluster.openshift.com/amd64/linux/oc.tar)
link for Download oc for Mac 200 OK (https://console-openshift-console.apps.bpetersen.devcluster.openshift.com/amd64/mac/oc.zip)
link for Download oc for Windows 200 OK (https://console-openshift-console.apps.bpetersen.devcluster.openshift.com/amd64/windows/oc.zip)
PASS
ok  	github.com/openshift/console-operator/test/e2e	5.274s
```

/assign @spadgett

### CODE REVIEW

File: test/e2e/downloads_test.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	framework.StandardCleanup(t, client)
+}
+
+func TestDownloadsEndpoint(t *testing.T) {
+	client, _ := setupDownloadsTestCase(t)
+	defer cleanupDownloadsTestCase(t, client)
+
+	route, err := client.Routes.Routes(api.OpenShiftConsoleNamespace).Get(api.OpenShiftConsoleName, v1.GetOptions{})
+	if err != nil {
+		t.Fatalf("error, can't get route: %s", err)
```
Comment (jhadvig) [inline test/e2e/downloads_test.go]: nit: would use the same string as we use on most of the places where we are erroring out on GET
```suggestion
		t.Fatalf("could not get route: %s", err)
```

Comment (benjaminapetersen) [reply test/e2e/downloads_test.go]: thx, updated!

**Comment by benjaminapetersen** [comment]
@jhadvig will need a new `lgtm` due to the `nit` fix.

**Comment by benjaminapetersen** [comment]
Ah, yes, I borrowed these funcs from the prev test, but moved them to `util_http.go` there so they could be shared with future tests:

```
test/e2e/metrics_test.go:110:6: httpOK redeclared in this block
previous declaration at test/e2e/util_http.go:53:34
test/e2e/metrics_test.go:117:6: getRequest redeclared in this block
previous declaration at test/e2e/util_http.go:20:44
```

**Comment by benjaminapetersen** [comment]
Didn't flag as an actual rebase issue.
This should pass pretty fast now.

**Comment by benjaminapetersen** [comment]
/retest

`TestRemoved` seems kinda flaky yet?

**Comment by benjaminapetersen** [comment]
Added a retry to the `TestRemoved()` e2e.  I think we need a bit of time to allow `terminated` resources to be `removed` for cleanup, else it looks like the operator is not working properly.

**Comment by benjaminapetersen** [comment]
@jhadvig @spadgett think we are good to go here.

**Pull request merged**